src/components/HTMLEngineProvider/HTMLRenderers/MentionReportRenderer.tsx

```diff
@@ -0,0 +1,125 @@
+        mentionDisplayText = report?.reportName ?? report?.displayName ?? htmlAttributeReportID;
+        // get mention details from name inside tnode
+    } else if ('data' in tnode && !isEmptyObject(tnode.data)) {
+        mentionDisplayText = removeLeadingLTRAndHash(tnode.data);
+
+        // eslint-disable-next-line rulesdir/prefer-early-return
+        Object.values(reports ?? {}).forEach((report) => {
+            if (report?.policyID === currentReport?.policyID && removeLeadingLTRAndHash(report?.reportName ?? '') === mentionDisplayText) {
+                reportID = report?.reportID;
+            }
+        });
+    } else {
+        return null;
+    }
+
+    return {reportID, mentionDisplayText};
+};
+
+function MentionReportRenderer({style, tnode, TDefaultRenderer, reports, ...defaultRendererProps}: MentionReportRendererProps) {
+    const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+    const htmlAttributeReportID = tnode.attributes.reportid;
+
+    const currentReportID = useCurrentReportID();
+    const currentReport = getReport(currentReportID?.currentReportID);
+    const isGroupPolicyReport = useMemo(() => (currentReport && !isEmptyObject(currentReport) ? ReportUtils.isGroupPolicy(currentReport) : false), [currentReport]);
+
+    const mentionDetails = getMentionDetails(htmlAttributeReportID, currentReport, reports, tnode);
+    if (!mentionDetails) {
+        return null;
+    }
+    const {reportID, mentionDisplayText} = mentionDetails;
+
+    const navigationRoute = reportID ? ROUTES.REPORT_WITH_ID.getRoute(reportID) : undefined;
+    const isCurrentRoomMention = reportID === currentReportID?.currentReportID;
+
+    const flattenStyle = StyleSheet.flatten(style as TextStyle);
+    const {color, ...styleWithoutColor} = flattenStyle;
+
+    return (
+        <ShowContextMenuContext.Consumer>
+            {() => (
+                <Text
+                    // eslint-disable-next-line react/jsx-props-no-spreading
+                    {...defaultRendererProps}
+                    style={
+                        isGroupPolicyReport
+                            ? [styles.link, styleWithoutColor, StyleUtils.getMentionStyle(isCurrentRoomMention), {color: StyleUtils.getMentionTextColor(isCurrentRoomMention)}]
+                            : []
+                    }
+                    suppressHighlighting
+                    onPress={
+                        navigationRoute && isGroupPolicyReport
+                            ? (event) => {
+                                  event.preventDefault();
+                                  Navigation.navigate(navigationRoute);
+                              }
+                            : undefined
+                    }
+                    role={isGroupPolicyReport ? CONST.ROLE.LINK : undefined}
+                    accessibilityLabel={isGroupPolicyReport ? `/${navigationRoute}` : undefined}
+                >
+                    #{mentionDisplayText}
```

Comment:
I think this is duplicating the `#` for `reportID` mentions.
<img width="339" alt="image" src="https://github.com/Expensify/App/assets/5201282/823676e7-c71d-4afa-b595-e3e3e287a2cb">